File: lib/http/cookie.go
package http

import (
	"net/http"

	"github.com/dop251/goja"
	"github.com/zengming00/go-server-js/lib"
)

type cookieRuntime struct {
	runtime *goja.Runtime
	cookie  *http.Cookie
}

func (This *cookieRuntime) stringFunc(call goja.FunctionCall) goja.Value {
	str := This.cookie.String()
	return This.runtime.ToValue(str)
}

func newCookie(runtime *goja.Runtime, cookie *http.Cookie) *goja.Object {
	obj := &cookieRuntime{
		runtime: runtime,
		cookie:  cookie,
	}
	o := runtime.NewObject()
	o.Set("string", obj.stringFunc)
	o.Set("domain", cookie.Domain)
	o.Set("expires", lib.NewTime(runtime, &cookie.Expires))
	o.Set("httpOnly", cookie.HttpOnly)
	o.Set("maxAge", cookie.MaxAge)
	o.Set("name", cookie.Name)
	o.Set("path", cookie.Path)
	o.Set("raw", cookie.Raw)
	o.Set("rawExpires", cookie.RawExpires)
	o.Set("secure", cookie.Secure)
	o.Set("unparsed", cookie.Unparsed)
	o.Set("value", cookie.Value)
	return o
}

File: js/http/cookie.js

var Name = 'test'
var Value = 'testval'
var Path = '/'
var MaxAge = 10
var HttpOnly = true


// response.setCookie(Name, Value, Path, MaxAge, HttpOnly)
response.setCookie(Name, Value, Path, MaxAge)
// response.setCookie(Name, Value, Path)


var cks = request.cookies()
var str = JSON.stringify(cks, null, 2)
log(str)

var r = request.cookie('test')
if (r.err) {
  throw r.err;
}
var cookie = r.value
log(cookie.name)
log(cookie.value)


function log(d) {
  console.log("%j", d);
  response.write(d);
  response.write('\r\n');
}

